client/ayon_core/pipeline/workfile/path_resolving.py

```diff
@@ -207,12 +214,94 @@ def get_workdir(
     )
 
 
-def get_last_workfile_with_version(
-    workdir, file_template, fill_data, extensions
-):
+@dataclass
+class WorkfileParsedData:
+    version: Optional[int] = None
+    comment: Optional[str] = None
+    ext: Optional[str] = None
+
+
+class WorkfileDataParser:
+    """Parse dynamic data from existing filenames based on template.
+
+    Args:
+        file_template (str): Workfile file template.
+        data (dict[str, Any]): Data to fill the template with.
+
+    """
+    def __init__(
+        self,
+        file_template: str,
+        data: dict[str, Any],
+    ):
+        data = copy.deepcopy(data)
+        file_template = str(file_template)
+        # Use placeholders that will never be in the filename
+        ext_replacement = "CIextID"
+        version_replacement = "CIversionID"
+        comment_replacement = "CIcommentID"
+        data["version"] = version_replacement
+        data["comment"] = comment_replacement
+        for pattern, replacement in (
+            # Replace `.{ext}` with `{ext}` so we are sure dot is not at the end
+            (r"\.?{ext}", ext_replacement),
+        ):
+            file_template = re.sub(pattern, replacement, file_template)
+
+        file_template = StringTemplate(file_template)
+        comment_template = re.escape(str(file_template.format_strict(data)))
+        data.pop("comment")
+        file_template = re.escape(str(file_template.format_strict(data)))
```

Comment:
What's this doing? Why pop the comment? Bit confused here.

Reply:
Added comment to the code, not sure if it helps?